src/mavoscript.js

```diff
@@ -533,6 +533,49 @@ var _ = Mavo.Script = {
 
 		return [a, b];
 	},
+	/**
+	 * Returns a list of all variables in the AST
+	 */
+	getVariables: function(ast) {
+		switch (ast.type) {
+			case "Literal":
+				return [];
+			case "Identifier":
+				// const isGlobal = !!globalThis[ast.name];
+				// return isGlobal? [] : [ast];	
+				return [ast];	
+			case "UnaryExpression":
+				return _.getVariables(ast.argument);
+			case "BinaryExpression":
+				if (ast.operator === ':') {
+					return _.getVariables(ast.right);
+				}
+				return [ast.left, ast.right].flatMap(_.getVariables);
+			case "LogicalExpression":
+				return [ast.left, ast.right].flatMap(_.getVariables);
+			case "CallExpression":
+				let toExplore = [...ast.arguments];
+				if (ast.callee.type !== "Identifier") {
+					toExplore = [ast.callee].concat(toExplore);
+				}
+				return toExplore.flatMap(_.getVariables);
+			case "Compound":
+				return ast.body.flatMap(_.getVariables);
+			case "ArrayExpression":
+				return ast.elements.flatMap(_.getVariables);
+			case "ConditionalExpression":
+				return [ast.test, ast.consequent, ast.alternate].flatMap(_.getVariables);
+			case "MemberExpression":
+				const children = _.getVariables(ast.object);
+				if (children[0] === ast.object) {
```

Comment:
Some comments about what this is doing and why would be useful.

Reply:
Sounds good, will add better, descriptive comments here (and same thing for below)!